components/ContentSearch/index.js

```diff
@@ -51,30 +53,37 @@ const ContentSearch = ({ onSelectItem, placeholder, label, contentTypes, mode, e
 		setSearchString(keyword);
```

Comment:
This may look like overkill and is totally optional for this PR, but once the search cache is converted to state, we can handle search changes a little more efficiently by responding to keyword changes with effect hooks.

```JS
/**
	 * handleSearchStringChange
	 *
	 * Using the keyword and the list of tags that are linked to the parent block
	 * search for posts/terms that match and return them to the autocomplete component.
	 *
	 * @param {string} keyword search query string
	 */
	const handleSearchStringChange = (keyword) => {
		setSearchString(keyword);
	}

	useEffect(() => {
		if (isLoading || searchString in searchCache) {
			return;
		}

		setIsLoading(true);

		const searchQuery = `wp/v2/search/?search=${searchString}&subtype=${contentTypes.join(
			',',
		)}&type=${mode}`;

		apiFetch({
			path: searchQuery,
		}).then((results) => {
			const newResults = results.filter((result) => {
				let keep = true;

				if (excludeItems.length) {
					excludeItems.forEach((item) => {
						if (item.id === result.id) {
							keep = false;
						}
					});
				}

				return keep;
			});

			setSearchCache(oldSearchCache => ({...oldSearchCache, [searchString]: newResults}));
			setIsLoading(false);
		});
	}, [isLoading, searchCache, searchString]);

	useEffect(() => {
		if (searchString in searchCache) {
			setSearchResults(searchCache[searchString])
		}
	}, [searchString, searchCache])
```

The first effect hook will run when the search term changes, when the cache changes, and when `isLoading` changes. This includes when `isLoading` becomes false -- guaranteeing that the latest search term will always queried. But by having that `isLoading` check at the start of the effect hook, we also ensure that only one fetch request is running   at a time.

The second effect hook responds to changes to the search cache. This will run after a successful `fetch` to populate the results.